nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/calcite/TestRecordResultSetOutputStreamCallback.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processors.standard.QueryRecord;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.serialization.RecordSetWriterFactory;
+import org.apache.nifi.serialization.SimpleRecordSchema;
+import org.apache.nifi.serialization.WriteResult;
+import org.apache.nifi.serialization.record.RecordField;
+import org.apache.nifi.serialization.record.RecordFieldType;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TestRecordResultSetOutputStreamCallback {
+
+    @Test
+    void testResultSetClosed() throws IOException, SQLException, ClassNotFoundException, InitializationException {
+        final TestRunner runner = TestRunners.newTestRunner(QueryRecord.class);
+
+        final String writerId = "record-writer";
+        runner.setProperty(writerId, writerId);
+
+        final RecordSetWriterFactory writerService = new CSVRecordSetWriter();
+        runner.addControllerService(writerId, writerService);
+        runner.setProperty(writerService, "schema-access-strategy", "inherit-record-schema");
+        runner.enableControllerService(writerService);
+
+        final ResultSet resultSet = getResultSet();
+
+        final RecordField fieldFirst = new RecordField("first", RecordFieldType.STRING.getDataType());
+        final RecordField fieldLast = new RecordField("last", RecordFieldType.STRING.getDataType());
+        final RecordSchema writerSchema = new SimpleRecordSchema(Arrays.asList(fieldFirst, fieldLast));
+
+        final FlowFile flowFile = mock(FlowFile.class);
+        when(flowFile.getAttributes()).thenReturn(new LinkedHashMap<>());
+
+        final AtomicReference<WriteResult> writeResultRef = new AtomicReference<>();
+        final AtomicReference<String> mimeTypeRef = new AtomicReference<>();
+
+        final RecordResultSetOutputStreamCallback writer = new RecordResultSetOutputStreamCallback(runner.getLogger(),
+                resultSet, writerSchema, 0, 0, writerService, flowFile.getAttributes(), writeResultRef, mimeTypeRef);
+
+        final OutputStream os = new ByteArrayOutputStream();
+        writer.process(os);
+
+        Assertions.assertTrue(resultSet.isClosed());
+    }
+
+    private ResultSet getResultSet() throws ClassNotFoundException, SQLException {
+        Class.forName("org.apache.calcite.jdbc.Driver");
```

Comment:
Instead of using the string class name, an actual instance of the Driver class could be passed to DriverManager.registerDriver()